src/core/Table/actionHandlers/selectHandler.ts

```diff
@@ -18,14 +18,81 @@ export const onSelectHandler = <T extends Record<string, unknown>>(
     return;
   }
 
-  const selectedData: T[] = [];
   const newSelectedRowIds = {} as Record<string, boolean>;
-  instance.initialRows.forEach((row) => {
-    if (newState.selectedRowIds[row.id] && !isRowDisabled?.(row.original)) {
+
+  const handleRow = (row: Row<T>) => {
+    if (isRowDisabled?.(row.original)) {
+      return false;
+    }
+
+    let isAllSubSelected = true;
+    row.initialSubRows.forEach((subRow) => {
+      const result = handleRow(subRow);
+      if (!result) {
+        isAllSubSelected = false;
+      }
+    });
+
+    // If `selectSubRows` is false, then no need to select sub-rows and just check current selection state.
+    // If a row doesn't have sub-rows then check its selection state.
+    // If it has sub-rows then check whether all of them are selected.
+    if (
+      (!instance.selectSubRows && newState.selectedRowIds[row.id]) ||
```

Comment:
Maybe any of you know how to write this if condition better?

Reply:
I think maybe leave like that together with the comments? Even tohugh some checks are repeated, but this way maybe will be clearer..

Reply:
I agree, it's fine with comments